app/controllers/password_resets_controller.rb

```diff
@@ -32,22 +32,28 @@ def update
   private
 
   def user_params
-    params.require(:user).permit :password, :password_confirmation
+    params.require(:user).permit(:password, :password_confirmation)
   end
 
   def get_user
     @user = User.find_by(email: params[:email])
+    return if @user
+
+    flash[:danger] = t "login_invalid"
+    redirect_to root_path
   end
 
   def valid_user
-    return if(@user && @user.activated? &&@user.authenticated?(:reset, params[:id]))
-    redirect_to root_url
+    if @user.authenticated?(:reset, params[:id])
+      return if @user.activated?
     end
+    redirect_to root_url
```

Comment:
Gộp code lại cho gọn gàng em nhé, mà nên thêm flash vô cho người ta biết lý do 
```suggestion
    return if @user.authenticated?(:reset, params[:id]) && @user.activated?
    
    flash ....
    redirect_to root_url
```